src/databricks/labs/ucx/source_code/path_lookup.py

```diff
@@ -30,14 +30,24 @@ def resolve(self, path: Path) -> Path | None:
         if path.is_absolute() and path.exists():
             # eliminate “..” components
             return path.resolve()
-        for parent in self.library_roots:
-            absolute_path = parent / path
+        for library_root in self.library_roots:
             try:
+                if not library_root.exists():
+                    continue
+
+                absolute_path = library_root / path
                 if absolute_path.exists():
-                    # eliminate “..” components
-                    return absolute_path.resolve()
+                    return absolute_path.resolve()  # eliminate “..” components
+
+                egg_subfolders = library_root.glob("*.egg")
```

Comment:
why do we need it here? it's too risky to do this change in so global place, please remove.

Reply:
> why do we need it here? it's too risky to do this change in so global place, please remove.

See [this comment](https://github.com/databrickslabs/ucx/pull/1789#discussion_r1618823311). I agree with this change being risky because it is in a global place. I added it here because it mimics Python is able to find libraries inside the `.egg` folders. And PathLookup should mimic Python's path lookup, right?

Reply:
> it mimics Python ability for finding libraries inside the .egg folders

okay, but add a comment with link to docs & add a unit test.